src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTest.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Formats.Tar;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "SdkContent")]
+public class ArtifactsSizeTest : SmokeTests
+{
+    private static readonly string BaselineFilePath = BaselineHelper.GetBaselineFilePath($"ArtifactsSizes/{Config.TargetRid}.txt");
+    private static readonly Dictionary<string, long> BaselineFileContent = new Dictionary<string, long>();
+    private static readonly Regex BuildVersionPattern = new(@"\b\d+\.\d+\.\d+[-@](alpha|preview|rc|rtm)\.\d(\.\d+\.\d+)?\b");
+    private const int SizeThresholdPercentage = 25;
+
+
+    public ArtifactsSizeTest(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        if (File.Exists(BaselineFilePath))
+        {
+            string[] baselineFileContent = File.ReadAllLines(BaselineFilePath);
+            foreach (string entry in baselineFileContent)
+            {
+                string[] splitEntry = entry.Split(':');
+                BaselineFileContent[splitEntry[0].Trim()] = long.Parse(splitEntry[1].Trim());
+            }
+        }
+        else
+        {
+            Assert.True(Directory.Exists(BaselineHelper.GetBaselineFilePath("ArtifactsSizes/")));
+            Assert.False(true, $"Baseline file `{BaselineFilePath}' does not exist. Please create the baseline file then rerun the test.");
+        }
+    }
+
+    [SkippableFact(new[] { Config.SourceBuiltArtifactsPathEnv, Config.SdkTarballPathEnv, Config.TargetRidEnv }, skipOnNullOrWhiteSpace: true)]
+    public void CompareArtifactsToBaseline()
+    {
+        Assert.NotNull(Config.SourceBuiltArtifactsPath);
+        Assert.NotNull(Config.SdkTarballPath);
+        Assert.NotNull(Config.TargetRid);
+
+        IEnumerable<TarEntry> artifactsTarEntries = Utilities.GetTarballContent(Config.SourceBuiltArtifactsPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+        IEnumerable<TarEntry> sdkTarEntries = Utilities.GetTarballContent(Config.SdkTarballPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+
+        (string FilePath, long Bytes)[] tarEntries = sdkTarEntries.Concat(artifactsTarEntries)
+            .Select(entry =>
+            {
+                string result = BaselineHelper.RemoveVersions(entry.Name);
```

Comment:
I noticed in the baselines that `rc` is appearing in a number of places.  There may be a bug in RemoveVersions in handling rc/preview version #s.

`Microsoft.DotNet.Common.ProjectTemplates.x.y.z.0.100-rc.x.y.z.nupkg: 147056`

Reply:
In that case, I think it would be necessary to detect duplicates and then just de-duplicate them by adding something like an incrementing integer as a suffix to the version.

Example:
```
SourceBuildReferencePackages/System.Threading.Thread.x.y.z.nupkg: 5264	
SourceBuildReferencePackages/System.Threading.Thread.x.y.z-1.nupkg: 5262
```

Reply:
The trick is to make sure all sorting is done with the original version value to maintain consistency across multiple builds.

Reply:
@MichaelSimons I've looked into the Regex that RemoveVersions uses, and it appears that it's using a Regex from this [semantic versioning definition site](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string). I tested it [here](regexr.com/7j47f), and the website is producing the same results as my test case. Given that this Regex came from this verifiable site (which was linked as a comment in BaselineHelper.RemoveVersions), I am unsure of whether I should be altering it for our purposes. What are your thoughts?

Reply:
Fixed it by adding in a Regex [here](https://github.com/dotnet/installer/pull/17233/commits/70cac71160d16ccb4bd41a271ae2e4f229faca9e#diff-2ed96e14569dd53766d8c841db49d63662b897532bc8e1bb9837f87e52476f12R107) and parsing the tarEntry name [here](https://github.com/dotnet/installer/pull/17233/commits/70cac71160d16ccb4bd41a271ae2e4f229faca9e#diff-ce0a9d7a18f02288a0510a8ec86e3a5f387ad96613e0296f434de93f02c18da7R87).